File: navigation/src/main/java/com/example/navigation/stepsEngine/field/AccountTypeField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.enums.AccountType;
import com.example.navigation.stepsEngine.payment.FlowState;

public class AccountTypeField extends Field {

    @Override
    public String get(FlowState flowState) {
        final AccountType accountType = flowState.getAccountType();
        if (accountType == null) {
            return null;
        } else {
            return accountType.name();
        }
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/AccountTypeField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState
import kotlinx.android.parcel.Parcelize

@Parcelize
class AccountTypeField : Field() {
    override operator fun get(flowState: FlowState): String? {
        val accountType = flowState.accountType
        return accountType?.name
    }

    override fun getId(): String {
        return FieldName.ACCOUNT_TYPE
    }

    override operator fun set(flowState: FlowState, value: Any?) {

    }
}